.pipelines/reusable-build.yml

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# This workflow performs a build of the project and uploads the result as a build artifact.
+
+parameters:
+  # Name associated with the output of this build.
+  - name: build_artifact
+    type: string
+  # Name of the job
+  - name: name
+    type: string
+  # Additional options passed to msbuild.
+  - name: build_options
+    type: string
+    default: ''
+  - name: generate_release_package
+    type: boolean
+    default: false
+  - name: build_codeql
+    type: boolean
+    default: false
+  - name: build_msi
+    type: boolean
+    default: false
+  - name: build_nuget
+    type: boolean
+    default: false
+  - name: cxx_flags
+    type: string
+    default: ''
+  - name: ld_flags
+    type: string
+    default: ''
+  - name: configurations
+    type: object
+    default: ["Debug", "Release"]
+
+
+jobs:
+  - job: '${{parameters.name}}'
+    timeoutInMinutes: 90
+
+    strategy:
+      matrix:
+        ${{ each configuration in parameters.configurations }}:
+          ${{ configuration }}:
+            buildConfiguration: ${{ configuration }}
+
+    pool:
+      vmImage: 'windows-2022'
+      type: windows
+
+    variables:
+      # Path to the solution file relative to the root of the project.
+      SOLUTION_FILE_PATH: ebpf-for-windows.sln
+      BUILD_ARTIFACT_NAME: ${{parameters.build_artifact}}
+      BUILD_CONFIGURATION: $(buildConfiguration)
+      BUILD_PLATFORM: x64
+      BUILD_OPTIONS: ${{parameters.build_options}}
+      CXX_FLAGS: ${{parameters.cxx_flags}}
+      LD_FLAGS: ${{parameters.ld_flags}}
+      MSBUILD_PATH: 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64'
+      GDN_CODESIGN_TARGETDIRECTORY: '$(Build.SourcesDirectory)/$(BUILD_PLATFORM)/$(buildConfiguration)'
+
+    steps:
+      - script: |
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```

Comment:
This line looks suspect since it assumes the path at which VS2022 is installed.  Seems like this should reference an environment variable instead.